test/jasmine/tests/layout_images_test.js

```diff
@@ -0,0 +1,217 @@
+
+            it('should work for right bottom', function() {
+                testAspectRatio('right', 'bottom', undefined, 'xMaxYMax');
+            });
+
+            it('should work for stretch sizing', function() {
+                testAspectRatio('middle', 'center', 'stretch', 'none');
+            });
+
+            it('should work for fill sizing', function() {
+                testAspectRatio('invalid', 'invalid', 'fill', 'xMinYMin slice');
+            });
+
+        });
+
+    });
+
+    describe('when the plot is dragged', function() {
+        var gd,
+            data = [{ x: [1,2,3], y: [1,2,3] }];
+
+        beforeEach(function() {
+            gd = createGraphDiv();
+        });
+
+        afterEach(destroyGraphDiv);
+
+        it('should not move when referencing the paper', function(done) {
+            var source = 'http://www.placekitten.com/200',
+                image = {
+                    source: source,
+                    xref: 'paper',
+                    yref: 'paper',
+                    x: 0,
+                    y: 0,
+                    width: 0.1,
+                    height: 0.1
+                };
+
+            Plotly.plot(gd, data, {
+                images: [image],
+                dragmode: 'pan',
+                width: 600,
+                height: 400
+            }).then(function() {
+                var img = Plotly.d3.select('[href="' + source + '"]').node(),
+                    oldPos = img.getBoundingClientRect();
+
+                mouseEvent('mousedown', 250, 200);
+                mouseEvent('mousemove', 300, 250);
+
+                var newPos = img.getBoundingClientRect();
+
+                expect(newPos.left).toBe(oldPos.left);
+                expect(newPos.top).toBe(oldPos.top);
+
+                mouseEvent('mouseup', 300, 250);
+            }).then(done);
+        });
+
+        it('should move when referencing axes', function(done) {
+            var source = 'http://www.placekitten.com/200',
+                image = {
+                    source: source,
+                    xref: 'x',
+                    yref: 'y',
+                    x: 2,
+                    y: 2,
+                    width: 1,
+                    height: 1
+                };
+
+            Plotly.plot(gd, data, {
+                images: [image],
+                dragmode: 'pan',
+                width: 600,
+                height: 400
+            }).then(function() {
+                var img = Plotly.d3.select('[href="' + source + '"]').node(),
+                    oldPos = img.getBoundingClientRect();
+
+                mouseEvent('mousedown', 250, 200);
+                mouseEvent('mousemove', 300, 250);
+
+                var newPos = img.getBoundingClientRect();
+
+                expect(newPos.left).toBe(oldPos.left + 50);
+                expect(newPos.top).toBe(oldPos.top + 50);
+
+                mouseEvent('mouseup', 300, 250);
+            }).then(done);
+        });
+
+    });
```

Comment:
Great tests so far. 

Let's make sure that `relayout` works as expected in a similar way to [`shapes`](https://github.com/plotly/plotly.js/blob/df5f58ed82c7f208da3058d5276466234f3eb2af/test/jasmine/tests/shapes_test.js#L160).